logstash-core/src/main/java/org/logstash/config/ir/CompiledPipeline.java

```diff
@@ -0,0 +1,388 @@
+                    if (ifvert.getOutgoingBooleanEdgesByType(true).stream()
+                        .anyMatch(edge -> Objects.equals(edge.getTo(), start))) {
+                        return splitLeft(newparents, iff, index);
+                    } else {
+                        return splitRight(newparents, iff, index);
+                    }
+                }
+            }).collect(Collectors.toList());
+    }
+
+    /**
+     * Build a {@link Dataset} representing the {@link JrubyEventExtLibrary.RubyEvent}s after
+     * the application of the given filter.
+     * @param vertex Vertex Id of the filter to create this {@link Dataset} for
+     * @param parents All the parent nodes that go into this filter
+     * @return Filter {@link Dataset}
+     */
+    private Dataset filterDataset(final String vertex, final Collection<Dataset> parents) {
+        final Dataset filter;
+        if (plugins.containsKey(vertex)) {
+            filter = plugins.get(vertex);
+        } else {
+            final RubyIntegration.Filter ruby = filters.get(vertex);
+            if (ruby.hasFlush()) {
+                if (ruby.periodicFlush()) {
+                    filter = new Dataset.FilteredFlushableDataset(parents, ruby);
+                } else {
+                    filter = new Dataset.FilteredShutdownFlushableDataset(parents, ruby);
+                }
+            } else {
+                filter = new Dataset.FilteredDataset(parents, ruby);
+            }
+            plugins.put(vertex, filter);
+        }
+        return filter;
+    }
+
+    /**
+     * Build a {@link Dataset} representing the {@link JrubyEventExtLibrary.RubyEvent}s after
+     * the application of the given output.
+     * @param vertex Vertex Id of the filter to create this {@link Dataset} for
+     * filter node in the topology once
+     * @param parents All the parent nodes that go into this output
+     * @return Output {@link Dataset}
+     */
+    private Dataset outputDataset(final String vertex, final Collection<Dataset> parents) {
+        final Dataset output;
+        if (plugins.containsKey(vertex)) {
+            output = plugins.get(vertex);
+        } else {
+            final RubyIntegration.Output ruby = outputs.get(vertex);
+            output = new Dataset.OutputDataset(parents, ruby);
+            plugins.put(vertex, output);
+        }
+        return output;
+    }
+
+    /**
+     * Split the parent {@link Dataset}s and return the dataset half of their elements that contains
+     * the {@link JrubyEventExtLibrary.RubyEvent} that <strong>do not</strong> fulfil the given
+     * {@link EventCondition}.
+     * @param parents Datasets to split
+     * @param condition Condition that must be not be fulfilled
+     * @param index Vertex id to cache the resulting {@link Dataset} under
+     * @return The half of the datasets contents that does not fulfil the condition
+     */
+    private Dataset splitRight(final Collection<Dataset> parents,
```

Comment:
Can we use a more semantic term than `split(Left|Right)`? Maybe `split(True|False)` instead?

Reply:
@andrewvc isn't "left" and "right" generally understood as true and false. It's used in books https://books.google.de/books?id=u5wkJ4R03d4C&pg=PA261&lpg=PA261&dq=left+branch+conditional&source=bl&ots=2iEMK4ma-3&sig=rO5OhVu1Cs9wyxq4cfu5NWYhEIo&hl=en&sa=X&ved=0ahUKEwj2gOT906XWAhXOh7QKHYBYDssQ6AEIWDAJ#v=onepage&q=left%20branch%20conditional&f=false I didn't make it up :D But sure I can rename that to true and false if it isn't :)